src/EditorFeatures/Core/GoToDefinition/IGoToDefinitionItemService.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.GoToDefinition
+{
+    internal interface IGoToDefinitionItemService : ILanguageService
+    {
+        Task<GoToDefinitionContext> GetDefinitionsAsync(
+            Document document,
+            int position,
+            bool thirdPartyNavigationAllowed,
+            CancellationToken cancellationToken);
+    }
+
+    internal struct GoToDefinitionContext
```

Comment:
in this case, this would be more of a result.  we think of the context objects as things we "push" the results into.  i.e. CompletionContext, or the contexts used in code fixes.